Assets/MixedRealityToolkit/Providers/BaseDeviceManager.cs

```diff
@@ -14,7 +14,7 @@ namespace Microsoft.MixedReality.Toolkit.Core.Providers
     /// <summary>
     /// Base Device manager to inherit from.
     /// </summary>
-    public class BaseDeviceManager : BaseExtensionService, IMixedRealityDeviceManager
+    public abstract class BaseDeviceManager : BaseExtensionService, IMixedRealityDeviceManager
```

Comment:
Why change the base implementation to an Abstract. Shouldn't do this unless there is a concrete reason as Abstraction adds a small but noticeable overhead.

Reply:
@SimonDarksideJ would you be so kind as to post a link or something confirming using abstract inheritance is slower than normal inheritance? I couldn't find anything about it.

Reply:
> Overall this seems sensible but needs a full round of performance validation and testing.

@SimonDarksideJ I agree here clearly. From what I can tell there are quite a couple of calls that are meant to alter the position of things. This is getting tricky when talking about let's say mouse delta, or scroll delta.
I guess someone would need to go through there and filter by what is relevant for the raycasting phase to be correct for all input devices. If that's even all there is to it.